### ISSUE TITLE
KTNE web version - 3 simple modules

### ISSUE THREAD

**Issue by azureus89** [224-1/xtnt-micro#46]
@Mazumda finalise 3 modules
draw and share the diagram here

@ParikshithMohan from your side, start with ideas for 3 modules -- keeping it simple for the mobile/web version of KTNE

it will not have a bomb theme, that is scaring away corporates, we will build it so that it can be skinned for various themes, so very functional abstract basic build ... objects will all be SVG and reactive/reagent driven ... when we get to this either me or baishakhi will assist you

lets start with a discussion here in this thread -- what 3 modules will you do
@Mazumda lets finalise and wrap this part up by EOD

**Comment by azureus89** [comment]
@Mazumda @ParikshithMohan soon

**Comment by ParikshithMohan** [comment]
I was thinking of one of those psychology games that Aarjavi showed me. Where the color displayed and the color written in words is different. We need to memorize the color thats written in words. This can be combined with a Simon-says like gameplay.

**Comment by ParikshithMohan** [comment]
Something like this : https://www.youtube.com/watch?v=OrMjwvcoeJY

**Comment by Mazumda** [comment]
am trying to dig out a milanote board gimme 5 min

**Comment by ParikshithMohan** [comment]
This game is brilliant too:  https://www.gamesforthebrain.com/game/guesscolors/

**Comment by Mazumda** [comment]
https://app.milanote.com/1I8G1a1JDCTP6V

**Comment by Mazumda** [comment]
i wanted to keep the modules mathematical, given that most workspaces have quant based ppl

**Comment by ParikshithMohan** [comment]
But is everybody good at math (or like math)? Some people are good at puzzles and problem solving too (and find it enjoyable even if its tricky).

**Comment by Mazumda** [comment]
i wanted some general rules

Each  module will have its own serial Number
Each  Module will have its own sub  timer 
Timer can be stopped for 10 seconds upto 5 times to strategize
Unlimited Strikes

**Comment by Mazumda** [comment]
there will be brownie points to have the least execution time, for each module.

**Comment by Mazumda** [comment]
i want to promote think more and execute less.

**Comment by azureus89** [comment]
will fork xtnt-micro to a new repo just for the game by EOD, just finish the discussion here, development will be on a new repo and bot updates will go to forge channel, for now too much reconfiguration for no added benefit while this discussion finishes

all dev work will be on that new repo

**Comment by Mazumda** [comment]
General Rules
The puzzle board is made up of multiple modules, the main board  has a sequence number and each module has its own sequence number. These 2 sequence numbers have to be matched in different ways specific to the module rule
Module serial numbers will digitally open into the manual ( taking out time required to call out module names etc)
 Each module has its own timer, however the module specific timer does not go off, unless you click and activate the module. the defuser should be able to see the timer specific to his module
All Players in the team should be able to concurrently view the game being played
Defusing players will be able to punch the  serial number and the module serial number to pull up the manual equivalent digitally
if there are more than 2 mistakes in any module, it will reset to its original state.
There needs to be a notification type feed that traces who did what when 
The game play will be of a fixed time interval, there can be as many strikes
Like tetris there can be a choice of unlimited modules, as modules get solved new modules spawn out, modules that are not successfully defused dont disappear
Handler  can right click and assign a module to a player in the team, allocation does not start the module timer, only one module can be allocated per defuser, defuser can give up on a module and move to the next one, number of give ups can be limited to X
the board can be shared like google docs in view format only for others to be able to see whats happening
Teams will be able to see the group goal / achievement

**Comment by Mazumda** [comment]
of the 3 modules, there should be one that is asynchronous and independent. stuff that you can explain and ppl will pick up on their own, Wires are an example of this. there should be one that is sychronous where 2 ppl need to actively solve,  i would go with  2 asynchronous and one synchronous module.

**Comment by Mazumda** [comment]
or maybe one semi synchronous, where the mistake count or a dynamic piece of information may or maynot change the rules, this is the case with simon says

**Comment by Mazumda** [comment]
parikshith to your fun comment, i was thinking of collaborating to replicate a drawing.

**Comment by Mazumda** [comment]
assume that its on a grid with co-ordinate positions available

**Comment by Mazumda** [comment]
and the challenge is to communicate the pattern, there are a fixed set of hints that one can use from a pool of hints.

**Comment by Mazumda** [comment]
or a fixed set of patterns that needed to be added or subtracted to arrive at the derived pattern

**Comment by azureus89** [comment]
@Mazumda my friend, pleaset try to collapse comments into one, instead of chat style, more like mail thread

**Comment by Mazumda** [comment]
https://app.milanote.com/1I8sIN1JDCTP6P 

added everything into the milanote, have a set of general rules that apply across modules and a set of samples

**Comment by azureus89** [comment]
If generic rules then can make independent form KTNE -- business objectives, pitfalls, goals -- so that for future background of playing KTNE is not there -- generalise this into our game design principles

**Comment by azureus89** [comment]
What's next here .. @Mazumda you had planned for documentation in 3 stages .. Parikshith had sent stage 1 and had approved .. move through next 2 stages as you had planned soon .. we should try to start work by Friday .. agree on modules and principles by then .. and I will make choice of graphic library or lack of it accordingly and work will commence on new repo Friday .. please progress

**Comment by azureus89** [comment]
@ParikshithMohan sorry for the complication
this is the ONLY branch for this issue ... we will not be creating an new issue for part 1 / part 2 / part 3

the mistake here was that all the issue modification happened in chat, not here, this was my mistake, it wont happen again

the new expectations etc will be added here

**Comment by azureus89** [comment]
@ParikshithMohan I had left the original PR open for you to complete all the documentation required in pk-46 ... submit commits, get commit level approval and then close PR when the whole issue is done ... @Mazumda is not familiar with this, you've done this before .... his role on this issue is restricted to helping you get on the same page as him, and understand the business interests of the game ...
do not rush / feel any extra pressure .. stick to the plan ... stick to the devflow you have always followed .. sorry, it wont happen again ...
this issue has 3 sub-tasks ... like many issues yio have completed before ...
rupam will define the sub tasks
commit the outputs in org files, i will approve the individual commits
once everything thats required for this issue is done, we will close the SINGLE pull request

**Comment by azureus89** [comment]
@Mazumda everything to do with finishing this issue and finalising 3 simple modules comes here ... when original plan changes, that is also captured here, in this issue
parikshith has a branch where all the code for dealing with this issue goes ... that branch has a pull request which i will close when the whole issue is completed, or occassionally when somebody else needs a part of the code in order to proceed their issue
the new re-definition of the task must be captured here in this issue, as a comment -- we used to edit the original issue description but we dont, because there is no way to recover history... instead update in comments ...
so lets start with this ... what is the new sub task breakdown for @ParikshithMohan 
please add to this comment thread with that
I know parikshith has finished a lot of the work, thats good, this is just for the record
@ParikshithMohan when @Mazumda has done that, please re-submit the files in your pk-46 branchm, and open a new PR
we will proceed forward from there

**Comment by azureus89** [comment]
Discussion about modules here or in the commit thread

**Comment by azureus89** [comment]
> from a user input perspective, team leads and sub teams ( pairs ) before each round should be able to provide an input on their preferred strategy.
> 
> in the Max score setting, the team commits to getting the module that has the max point potential, this would obviously mean that modules have different score weights and will be more applicable when there are many modules and module variants to choose from.
> 
> In the User pref setting, modules need to roll up to module types, in our case it is sync / Async. when we used to play KTNE people would shout out their pref on what they wanted, as team lead, i would often consider / discount their inputs depending on what modules got served. In our case users can indicate a pref, which the team lead may or may not consider.
> 
> Utilization based is competely meta data driven and comes into play when deciding which player to allocate the module to. In the attached sheet i have attempted to draw out how the selection would work from an automated algorithm perspective
> https://docs.google.com/spreadsheets/d/1I7nDDN98o6JFUdcgeevst-BkZh2dAymWd0eyO7j4ZtU/edit?usp=sharing
> it should be possible for a strategy to back fire, under utilize team etc it will consider
> a. Unallocated time
> b. Self rating on how comforable you are with the game as a player before a round starts, this could also be module specific and be applicable post the first 2 rounds of play ( maybe a single slider where the player can say if they played the game before or if they are comfortable with games and so on)
> c. The best team / pair who scores the max can be the preferred team from the pool of waiting teams from an allocation perspective, this will require the stats to be available as an input, and the scoring mechanism to be fleshed out first
> 
> maybe a better idea is to baseline names of measures and then define as an expression what a KPI means. measures are just measures. for example
> 
> User module pref ( upto 3 pref)
> pref 1
> pref 2
> pref 3
> 
> Waiting time - Time that a team spends un allocated to a module
> Avg Module solving time - Avg time of solving a module
> Allocation of a module -
> option 1 select team with max waiting time.
> option 2 Select team with min avg solving time of module
> 
> in this way the KPI defination can be completely clear. let me see if i can come up with the data definition of the measures and the calculations of the KPIs so that it doesnt confuse people.

moving this discussion here, because it seems unrelated to commit thread about priority on data metrics .. lot of these seem like good to haves not urgent priorities to get the game going by nov 15th

im going through what you are saying, and just seeing if im capturing enough for us to pull these views on the data later - either to build features or analytics on top of  ... seems okay because we have a graph db kind of structure in the back ... and mostly you are talking about relationships ... and things you could build on relationships ... 
find x person (or x team or x & y person join) -> find modules attempted by x -> filter solved -> avg time
but need to prioritise some things
some of these from a feature standpoint are like waaay down the pipeline

I somewhat understand what you are talking about here tbh ... but would prefer if you rephrase when you have the time if this has to go into the framework ... the phrasing and communication is not very clear -- suggest simplify / rephrase ... since you need you to funnel through parikshith, so that whatever is part of the framework is captured in the org files ... and priority stuff highlighted

you can use any format/medium you want to explain to parikshith, onus is on him to give me documentation in the format I want

**Comment by azureus89** [comment]
@Mazumda @ParikshithMohan updated single reference doc - easy for humans to read here on my branch https://github.com/224-1/xtnt-micro/blob/az-46-updated-data-metrics/doc/forge/ktne/design-principles.org

@ParikshithMohan submitted PR, please merge when you can, unless you want to rewrite
and please work with @Mazumda to highlight priority metrics in the Data in Forge games section

ASAP

Edited to change the link to the renamed file

**Comment by ParikshithMohan** [comment]
**Module-1** 
MAZE
The handler is presented with a maze similar to ktne (just dots, no obstructions). But here there are only two symbols - red and green. The catch is that the symbols look different and its the job of the handler to describe exactly what the symbols look like, to the defuser. The defuser has six different mazes with different red and green symbols. Based on the information given by the handler, the defuser has to pick one of the six mazes and help the handler move the red symbol to the green. 
Strikes: 3 max
![maze](https://user-images.githubusercontent.com/24820582/68071681-b386dc80-fda2-11e9-9544-c7ea6a5f6143.png)

**Comment by ParikshithMohan** [comment]
**Module-2**
**Objective:** To crack the code using alphabets and symbols.
**Handler:** The handler's screen has 3 elements. There is a row of 8 boxes with numbers between the range 1 to 26, written inside them. There is another set of eight boxes that are empty. The handler has to type in letters to crack the code. The third element is a pinwheel, which has eight symbols in it. This pinwheel rotates when a button is pressed. Now its the handler's job to describe the symbol to the handler when the pin lands on any one of the symbols and also to read out the number written in the boxes. 
![mod2-1](https://user-images.githubusercontent.com/24820582/68072163-6279e700-fda8-11e9-8c11-d063fff5a344.png)

**Defuser:** The defuser notes down the numbers and has to match them to their respective alphabets. For example: 3 = C . Next, when the defuser understands the symbol, he has to find the value associated with it. so if its a pumpkin the value associated with it is +3, next the defuser has to add 3 to each number and rematch the new numbers to the respective alphabets. Once the answer is found, the defuser has to read it out to the handler. The handler can then type in the letters in the empty boxes. The module is completed when the handler types in the correct code.
![mod2-2](https://user-images.githubusercontent.com/24820582/68072332-90602b00-fdaa-11e9-9a69-8476c87b07fe.png)
**THE ANSWER TO THE ABOVE PROBLEM IS: P F M W A D T V**

**Comment by ParikshithMohan** [comment]
**Module 3**
MORSE CODE
Same as ktne morse code but much more simple. The handler is presented with 3 bulbs (yellow, blue, green), a frequency meter and a button. The yellow bulb gives out dots and dashes, the blue bulb blinks to denote the end of a word and the green bulb shows if the module has been cleared. 
The handler has to note down the dots & dashes and has to report it to the defuser. The defuser has to combine all the information received and form a word. Then find the frequency linked to that word and report it back to the handler. The handler can then set the frequency in the meter and press the 'go' button. If the frequency is right, the green bulb will glow, else all the bulbs turn red. 
strikes:3max

![mod 3-1](https://user-images.githubusercontent.com/24820582/68073401-e25a7e00-fdb5-11e9-94e5-9d98c21e47f9.png)
![mod3-2](https://user-images.githubusercontent.com/24820582/68073402-e25a7e00-fdb5-11e9-8533-86d128f34d8b.png)

**Comment by azureus89** [comment]
going ahead with above 3 modules as first 3 simple modules, closing issue
